AUTH.md

```diff
@@ -133,3 +134,40 @@ NEXT_PUBLIC_AZURE_AD_URL_WITSMLEXPLORER=http://localhost:3000/
 NEXT_PUBLIC_AZURE_AD_SCOPE_API=
 
 ```
+
+## Hybrid Flow
+It is now possible to utilize both OAuth2 and Basic authentication security schemes at the same time. With this configuration the end user will have to authenticate against an authorization server that supports `OAuth2 Authorization Code Flow with PKCE`.
+
+As described in the serverlist, Witsml-Explorer will choose the correct method to contact the backend, based on how you configure your list of servers. If the Server has securityscheme `OAuth2` and both the server and user have the same `app-roles`, the frontend will use the `Bearer` token received from the authorization server and relay this to the backend along with the url of the server. It will not ask the user for basic credentials (username:password).
+
+The backend in turn will do a lookup on the users application role, and if eligible, use the `system-user` fetched from `keyvault` for further connections to the WITSML server.
+
+### Basic witsmlexplorer flow
+For an illustration of sequences initiated when the user queries a server with property securityscheme set to `Basic`, visit the figure in the beginning of this document under `WITSML server credentials flow`
+
+### OAuth2 witsmlexplorer flow
+The following diagram illustrates the flow for a user contacting a `Server` that have been assigned securityscheme `OAuth2` (Azure example):
+
+```mermaid
+
+graph TD
+    A[1. WitsmlExplorer Frontend] -->|login| B[2. Auth Server]
+    B[2. Auth Server] --> |token| A[WitsmlExplorer Frontend]
+    A[1. WitsmlExplorer Frontend] --->|getservers| G[3. ServerlistDB]
```

Comment:
Should it be mentioned that with msalEnabled, a token is needed to fetch the servers as well?

Reply:
good point! CRUD operations on the serverlist will have to be done by an authenticated user. I will adjust the documentation and figure.